fdbserver/GrvProxyServer.actor.cpp

```diff
@@ -1134,10 +1137,12 @@ ACTOR Future<Void> grvProxyServer(GrvProxyInterface proxy,
 		wait(core || checkRemoved(db, req.recoveryCount, proxy));
 	} catch (Error& e) {
 		TraceEvent("GrvProxyTerminated", proxy.id()).errorUnsuppressed(e);
-
+		CODE_PROBE(e.code() == error_code_master_failed, "master_failed");
```

Comment:
Should we assert here that we can't get a `broken_promise`?

Reply:
I think it would be good idea to put an assert and re-run the batter of test to make sure all cases are covered

Reply:
a nit: worth adding better CODE_PROBE statement, maybe `GrvProxyServer master failed"` or something similar?

Reply:
have added the assert and rerun the test and there are no assert failures